src/ui/components/Library/Team/View/TimeFilterContextRoot.tsx

```diff
@@ -0,0 +1,53 @@
+import { Dispatch, ReactNode, SetStateAction, createContext, useMemo, useState } from "react";
+
+export type TimeFilterOptions = "hour" | "day" | "week" | "two-week" | "month";
+
+type TimeFilterContextType = {
+  filterByTime: TimeFilterOptions;
+  startTime: Date;
+  endTime: Date;
+  setFilterByTime: Dispatch<SetStateAction<TimeFilterOptions>>;
+};
+
+export const TimeFilterContext = createContext<TimeFilterContextType>(null as any);
+
+const daysAgo = (days: number) => {
+  const date = new Date();
+  date.setDate(date.getDate() - days);
+  // Make end time consistent
+  if (days > 1) {
+    date.setHours(0, 0, 0, 0);
+  } else {
+    date.setMinutes(0, 0, 0);
+  }
```

Comment:
@ryanjduffy Does this seem reasonable to you? For example if current time 10:43, and we filter by last hour, it'll make a request for 10:00 - 10:43.

I need to double check what happens if we are in time like 10:01. This probably isn't right.

Reply:
IMO if it says e.g. "this month" then it makes sense to start on the first day of the month but if it says "last month" then it should be the last ~30.5 days.

In this case, since we say "last", then I think we should track 60 minutes from the current time (to use your example).

Reply:
@bvaughn Thanks! I think the best phrasing is "Last 30 days" because that's as clear as possible.

Reply:
* If you select "This month" on December 31: Dec 1-31
* If you select "This month" on December 1: Also Dec 1-31

* If you select "Last 30 days" on December 31: Dec 1-31, because that's 30 days from current day
* If you select "Last 30 days" on December 1: November 2-December 1 (whatever is 30 days earlier from current day, calendars are confusing)

Reply:
cc @bvaughn @callingmedic911